tests/test_uploads.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+'''Tests for image upload API endpoint.'''
+
+import os.path
+import sys
+import unittest
+
+from flask import Request
+from StringIO import StringIO
+from werkzeug import FileStorage
+from werkzeug.datastructures import MultiDict
+
+# Add webcompat module to import path
+sys.path.append(os.path.realpath(os.pardir))
+
+from webcompat import app
+
+
+class TestUploads(unittest.TestCase):
+    # modified from http://prschmid.blogspot.com/2013/05/unit-testing-flask-file-uploads-without.html
+    def setUp(self):
+        app.config['TESTING'] = True
+        self.app = app
+        self.test_client = self.app.test_client()
+
+    def tearDown(self):
+        pass
+
+    def testGet(self):
+        '''Test that /upload/ doesn't let you GET.'''
+        rv = self.test_client.get('/upload/')
+        self.assertEqual(rv.status_code, 404)
+
+    def testBadUploads(self):
+        # Loop over some files and the status codes that we are expecting
+        for filename, status_code in \
+                (('foo.xxx', 415), ('foo', 415), ('foo.rb', 415)):
+
+            # The reason why we are defining it in here and not outside
+            # this method is that we are setting the filename of the
+            # TestingFileStorage to be the one in the for loop. This way
+            # we can ensure that the filename that we are "uploading"
+            # is the same as the one being used by the application
+            class TestingRequest(Request):
+                """A testing request to use that will return a
+                TestingFileStorage to test the uploading."""
+                @property
+                def files(self):
+                    d = MultiDict()
+                    d['image'] = TestingFileStorage(filename=filename)
+                    return d
+
```

Comment:
Ah no this will not solve your issue. So what the place you copied from is trying to redefine locally a property of a class.

I was looking at the source of `~/.virtualenvs/webcompatcom/lib/python2.7/site-packages/flaskext/uploads.py` which is already patching partly app.request class.

**Let it leave like you initially propose.**

 I'll try to find a more elegant solution later. This one is working, even if not very DRY. I have the feeling that with `super()` we should be able to override a bit things in the same way that he wanted to do it.